src/source/Response.c

```diff
@@ -445,24 +472,25 @@ STATUS curlCompleteSync(PCurlResponse pCurlResponse)
 STATUS notifyDataAvailable(PCurlResponse pCurlResponse, UINT64 durationAvailable, UINT64 sizeAvailable)
 {
     STATUS retStatus = STATUS_SUCCESS;
-    CURLcode result;
+    CURLMcode result;
+    PCallbacksProvider pCallbacksProvider;
 
     CHK(pCurlResponse != NULL, STATUS_NULL_ARG);
 
+    if (pCurlResponse->pCurlRequest != NULL && pCurlResponse->pCurlRequest->pCurlApiCallbacks != NULL &&

```

Comment:
If any if these are NULL,`pCallbacksProvider` will be pointing to a random location, correct? Wouldn't this cause problems on LN 491?

Reply:
Yes you're right, I initialized them null and check to make sure it's not null before using in new revision

Reply:
@disa6302 if that's the case then all bets are off - you have a mem corruption